nltk/twitter/twitterclient.py

```diff
@@ -0,0 +1,457 @@
+    def sample(self):
+        """
+        Wrapper for 'statuses / sample' API call
+        """
+        while self.do_continue:
+            '''
+            Stream in an endless loop until limit is reached
+            see twython issue 288
+                https://github.com/ryanmcgrath/twython/issues/288
+                colditzjb commented on 9 Dec 2014
+            '''
+            try:
+                self.statuses.sample()
+            except requests.exceptions.ChunkedEncodingError as e:
+                if e is not None:
+                    print("Error (stream will continue): {0}".format(e))
+                continue
+
+    def filter(self, track='', follow='', lang='en', locations=None):
+        """
+        Wrapper for 'statuses / filter' API call
+        """
+        while self.do_continue:
+            '''
+            Stream in an endless loop until limit is reached
+            see twython issue 288
+                https://github.com/ryanmcgrath/twython/issues/288
+                colditzjb commented on 9 Dec 2014
+            '''
+            try:
+                if track == '' and follow == '' and locations == None:
+                    msg = "Please supply a value for 'track', 'follow' or 'locations'."
+                    raise ValueError(msg)
+                self.statuses.filter(track=track, follow=follow, lang=lang,
+                                     locations=locations)
+            except requests.exceptions.ChunkedEncodingError as e:
+                if e is not None:
+                    print("Error (stream will continue): {0}".format(e))
+                continue
+
+
+
+
+class Query(Twython):
+    """
+    Retrieve data from the Twitter REST API.
+    """
+    def __init__(self, app_key, app_secret, oauth_token,
+                 oauth_token_secret):
+        self.handler = None
+        self.do_continue = True
+        Twython.__init__(self, app_key, app_secret, oauth_token, oauth_token_secret)
+
+    def register(self, handler):
+        """
+        Register a method for handling tweets.
+
+        :param TweetHandlerI handler: method for viewing or writing tweets to a file.
+        """
+        self.handler = handler
+
+    def _lookup(self, infile, verbose=True):
+        """
+        :param infile: name of a file consisting of Tweet IDs, one to a line
+        :return: iterable of Tweet objects
+        """
+        with open(infile) as f:
+            ids = [line.rstrip() for line in f]
+        if verbose:
+            print("Counted {0} Tweet IDs in {1}.".format(len(ids), infile))
+
+        id_chunks = [ids[i:i+100] for i in range(0, len(ids), 100)]
+        """
+        The Twitter endpoint takes lists of up to 100 ids, so we chunk
+        the ids.
+        """
+
+        listoflists = [self.post('statuses/lookup', {'id': chunk}) for chunk
+                       in id_chunks]
+        return itertools.chain.from_iterable(listoflists)
+
+
+    def lookup(self, infile, outfile, verbose=True):
```

Comment:
This was introduced on the assumption that we would only be allowed to distribute Tweet IDs, not full Tweets. As we subsequently received a clarification from Twitter that we can in fact include up to 50k Tweets in the NLTK data distribution, the use case for `lookup()` is much less prominent, and I decided to remove the usage examples from the `twitter_demo.py` file. 

I'm in two minds about what to do now. It might be better packaged up as a standalone command line utility, since it probably won't form part of a Python processing pipeline. If it's a standalone utility, then I'd be inclined to leave it to the user to address compressing / decompressing file inputs and outputs.